pkg/labels/labels.go

```diff
@@ -59,10 +59,29 @@ func (ls Labels) String() string {
 		b.WriteString(strconv.Quote(l.Value))
 	}
 	b.WriteByte('}')
-
 	return b.String()
 }
 
+// Bytes returns ls as a byte slice.
+// It uses an byte invalid character as a separator and so should not be used for printing.
+func (ls Labels) Bytes(buf []byte) []byte {
```

Comment:
Relating to @roidelapluie's comments:

This function doesn't need to insert `{`, `}`, or `=` at all.

For empty labels, it doesn't need to return anything.

For the other cases, it returns, for each label, `sep + l.Name + sep + l.Value`. That should be unique for each possible unique label set.

Reply:
} is needed when we concatenant labels together, e.g. to see the difference between:

{a="b",b="c"}{c="d"}
{a="b"}{b="c",c="d"}

> a$b$b$c$c$d$

We need an extra separator:

> a$b$$b$c$c$d$
> a$b$b$c$$c$d$

Reply:
Even if we did, would it be a difference?

Reply:
> weelll I don't think we ever do this so this is not needed

We do exactly this in `resultMetric`. Keeping `{ ... }` to differentiate between lhs and rhs when concatenating in `resultMetric` but removing `, ` and `=` so that the bytes are just `{nameSEPvalueSEPnameSEPvalue}` for the main benchmark I usually run buys us a little bit of ns/op improvement but seems to both reduce and increase bytes/allocs.

Reply:
Then we have to do something else. `}` and `{` could be part of a label value. How about three separator bytes to separate lhs and rhs?

Reply:
Or another separator

Reply:
Keep in mind that label names cannot be empty.

Reply:
> Then we have to do something else. } and { could be part of a label value. How about three separator bytes to separate lhs and rhs?

> Or another separator

A different invalid character separator seems like the best option.

> Keep in mind that label names cannot be empty.

@brian-brazil are you point out that at this point label names are guaranteed to not be empty, or saying we need to check that names are not empty here? Just had a quick look but I don't see anywhere else we're checking for empty names. `labels.WithoutEmpty` only looks for empty values.

Reply:
My point is that could be a useful property for ensuring uniqueness when the strings are concatenated.

Reply:
Given that there are plenty of bytes that can never occur in a valid UTF-8 stream, I agree with @cstyan that just using another invalid byte is the easiest.